customreboot/menu.js

```diff
@@ -0,0 +1,166 @@
+class RebootSubMenu extends PopupMenu.PopupSubMenuMenuItem {
+
+    _init() {
+        super._init(__('Restart to…'));
+
+        // Add boot options to menu
+        try {
+            this.createBootMenu();
+        }
+        catch (e) {
+            LogWarning(e);
+        }
+    }
+
+    async createBootMenu() {
+        // Get boot options
+        const type = await Bootloader.GetUseableType();
+        console.log(`Using ${type}`)
+
+        const loader = await Bootloader.GetUseable(type);
+
+        this.section = new PopupMenu.PopupMenuSection();
+
+        this.menu.open = (animate) => {
+            const heightLimit = 250
+            if (this.menu.isOpen)
+                return;
+
+            if (this.menu.isEmpty())
+                return;
+
+            this.menu.isOpen = true;
+            this.menu.emit('open-state-changed')
+
+            this.menu.actor.show()
+
+            let targetAngle = this.menu.actor.text_direction === Clutter.TextDirection.RTL ? -90 : 90;
+            let [, naturalHeight] = this.section.actor.get_preferred_height(-1)
+            if (naturalHeight > heightLimit) {
+                animate = false
+                naturalHeight = heightLimit;
+                this.menu.actor.vscrollbar_policy = St.PolicyType.AUTOMATIC;
+                this.menu.actor.add_style_pseudo_class('scrolled')
+            } else {
+                this.menu.actor.vscrollbar_policy = St.PolicyType.NEVER;
+                this.menu.actor.remove_style_pseudo_class('scrolled')
+            }
+            const duration = animate ? 250 : 0;
+            this.menu.actor.height = 0
+            this.menu.actor.ease({
+                height: naturalHeight,
+                duration,
+                mode:Clutter.AnimationMode.EASE_OUT_EXPO,
+                onComplete: () => this.menu.actor.set_height(naturalHeight),
+            });
+            this.menu._arrow.ease({
+                rotation_angle_z: targetAngle,
+                duration,
+                mode: Clutter.AnimationMode.EASE_OUT_EXPO,
+            });
+        }
+
+        this.menu.addMenuItem(this.section)
+
+        if (loader === undefined) {
+            // Add reload option, to refresh extension menu without reloading GNOME or the extension
+            this.section.addMenuItem(new PopupMenu.PopupSeparatorMenuItem());
+            this.section.addAction('Reload', () => {
+                this.section.removeAll();
+                this.createBootMenu();
+            });
+
+            // Add button to open settings
+            this.section.addAction('Settings', () => {
+                Extension.lookupByUUID('hibernate-status@dromi').openPreferences();
+            });
+
+            return;
+        }
+
+        loader.GetBootOptions().then(([bootOps, defaultOpt]) => {
+            if (bootOps !== undefined) {
+                this._itemsSection = new PopupMenu.PopupMenuSection();
+
+                for (let [title, id] of bootOps) {
```

Comment:
`title` and `id` fields are swapped here. systemdBoot.js and efibootmgr.js sets the `id` as first parameter, and `title` is the second parameter.